client/src/App.js

```diff
@@ -24,16 +25,50 @@ const theme = createMuiTheme({
 });
 
 function App() {
-  const [authStatus, setAuthStatus] = useState(true);
+  const [authStatus, setAuthStatus] = useState(false);
+
+  // Declare state variables
+  let [user, setUser] = useState(null)
+
+  useEffect(() => {
+    decodeToken()
+  }, [])
+
+  const updateUser = newToken => {
+    if (newToken) {
+      localStorage.setItem('mernToken', newToken)
+      decodeToken(newToken);
+    } else {
+      setUser(null);
+    }
+  }
+
+  const decodeToken = existingToken => {
+    let token = existingToken || localStorage.getItem('mernToken');
+
+    if (token) {
+      let decoded = jwtDecode(token);
+      // Check for expired token or non valid or user is not logged in at all
+      if (!decoded || Date.now() >= decoded.exp * 1000) {
+        console.log("Token expired!!")
+        setUser(null);
+      } else {
+        setUser(decoded);
+      }
+    } else {
+      setUser(null);
+    }
+  }
+
   function setNav() {
-    return authStatus ? <PostAuthNav /> : <PreAuthNav />;
+    return authStatus ? <PostAuthNav updateUser={updateUser} user={user}/> : <PreAuthNav updateUser={updateUser} user={user}/>;
```

Comment:
I also don't think the Nav bars need the state of the user passed down to them, just the updateUser event.